### ISSUE TITLE
Add /approvals endpoint to citygram

### ISSUE THREAD

**Issue by bcipolli** [opensandiego/open-dsd-notifications#14]
Currently, we have code enforcement live at citygram. There is an `/approvals` endpoint in our node.js router. See: https://sd-notifications.herokuapp.com/dsd/approvals

@MrMaksimize and @ahempton What data is the `/approvals` endpoint using? Is it permit approvals? Are there any specific types of permits?

Looks like other cities using citygram separate by permit type (building, electrical, and land-use all have [icons at citygram](https://github.com/codeforamerica/citygram/tree/master/app/assets/img/publishers/icons).

**Comment by bcipolli** [comment]
Assigning to myself. Once I get some details, I can have a PR ready to citygram within an hour's worth of work. @ahempton if there are other endpoints you'd like to see done, we can push them through relatively easily, with a bit of knowledge on what the data are and where to access them. Thanks!

**Comment by ahempton** [comment]
Hi @bcipolli -- OK, this does get a little confusing, but essentially here is how it works.

There are Projects and then there are Approvals.  Approvals fall under Projects.  Projects can contain multiple Approvals.

Projects can be either Ministerial or Discretionary.  Ministerial can be thought of as "over the counter" permits and include construction-related approvals such as Building Permits, Right-of-Way Permits, Electrical Permits, Mechanical Permits, Combination Building Permits, and so on.

Discretionary projects are ones that require a higher permit process and level of review before an applicant can submit for ministerial construction permits.  Discretionary approvals include permits like a Neighborhood Use/Development Permit, Conditional Use Permit, Site Development Permit, Planned Development Permit, and so on.  Discretionary approvals involve a public notice and the decision to approve is made by either Staff, Hearing Officer, Planning Commission, or City Council.

You can see here http://www.sandiego.gov/development-services/opendsd/index.shtml that there's a map to browse Ministerial or Discretionary approvals.

I think the Projects end-point might be more useful to have listed, since it includes all types of approvals -- in the future it would be nice to be able to narrow down by approval type (electrical, building, etc.) -- but letting people get alerts at the Project level would be useful because it would then relate to all approval types.

Let me know if you have any questions?

**Comment by bcipolli** [comment]
got it @ahempton . Any idea what feed would be good for projects? Approval data is from here: http://opendsd.sandiego.gov/web/Api/Maps/ApprovalsMap?SearchType=Discretionary&SouthWestLatitude=32.71879985593221&SouthWestLongitude=-117.16525563507082&NorthEastLatitude=32.74399836325726&NorthEastLongitude=-117.12534436492922&Page=1&PageLimit=100

I also don't see where the endpoints for `http://opendsd.sandiego.gov/web/Api/Maps` are documented. The only API doc I found at the OpenDSD webpage is: http://www.sandiego.gov/development-services/opendsd/pdf/opendsdapiv20141031.pdf

I can hack it up from here, but I assume there should be docs?
view-source:http://opendsd.sandiego.gov/web/maps/approvalsdiscretionary

**Comment by ahempton** [comment]
@bcipolli The only documentation I'm aware of is that PDF you're linking to above, but I agree there should be more details available.  I've sent an e-mail to someone on the development side and they may have some more details.  I might not hear back from them until Mon.

http://opendsd.sandiego.gov/api/project/:id

The above can't be used because it's not /api/maps right?

**Comment by bcipolli** [comment]
I believe that will just give me details about one project, rather than all projects. Looking over the different maps, I see there's only Approvals (one for discretionary, one for ministerial).

I think we should try to avoid publishing any feed that we want to change later, as Citygram doesn't really have that capability. It might frustrate subscribers for a feed to change or disappear.

Great to hear what else might be available (as far as APIs), then decide if exposing approvals (separate subscriptions for the two types, or one aggregated feed) is the right thing for users.

Thanks!

**Comment by ahempton** [comment]
Yes you're right Projects just brings up one project.  I see the problem.  Hopefully I can get some more details.

**Comment by ahempton** [comment]
@bcipolli I'm checking with the developers to see if there are more endpoints... so far they are saying no, except for some that were made for a hackathon... I told them about this project and am seeing if they might be available to chat with you for a couple mins./answer a couple questions

**Comment by bcipolli** [comment]
:+1:

**Comment by ahempton** [comment]
@bcipolli Ben I just sent an e-mail to your UCSD e-mail address with contact info for Kyle Smith with CGI who should be able to provide more details about the Open DSD specs.  Thanks for all your work on this.
